trl/trainer/grpo_trainer.py

```diff
@@ -195,8 +200,18 @@ def __init__(
                     "This argument can only be used when the `model` argument is a string."
                 )
 
+        # Track whether we're using LoRA and the rank of the LoRA adapter
+        lora_rank = 0
         if peft_config is not None:
             model = get_peft_model(model, peft_config)
+            if isinstance(peft_config, LoraConfig):
+                lora_rank = peft_config.r
```

Comment:
Just note that there could be different ranks for different layers: https://huggingface.co/docs/peft/v0.14.0/en/package_reference/lora#peft.LoraConfig.rank_pattern. Not sure if vLMM supports that or not. This might be relevant for determining the `max_lora_rank` below.